pyccel/codegen/printing/ccode.py

```diff
@@ -347,6 +348,8 @@ def is_c_pointer(self, a):
             a = a.funcdef.results[0].var
         if isinstance(getattr(a, 'dtype', None), CustomDataType) and a.is_argument:
             return True
+        if a.get_user_nodes(DottedVariable) and a.cls_base is self._current_class:
+            return True
```

Comment:
I don't understand this. If the object is used by a DottedVariable and the class base is the current class. Why is it necessary for one of the elements to be accessed? What if `self` is passed to another function?

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1472/commits/b9b353587bd848a608c5e2150cef6434decb65f0.